tests/ignite/contrib/metrics/test_cohen_kappa.py

```diff
@@ -0,0 +1,299 @@
+import os
+
+import numpy as np
+import pytest
+import torch
+from sklearn.metrics import cohen_kappa_score
+
+import ignite.distributed as idist
+from ignite.contrib.metrics import CohenKappa
+from ignite.engine import Engine
+from ignite.exceptions import NotComputableError
+
+
+def test_no_update():
+    ck = CohenKappa()
+
+    with pytest.raises(NotComputableError):
+        ck.compute()
+
+
+def test_wrong_inputs():
+    ck = CohenKappa()
+
+    with pytest.raises(ValueError):
+        # y has not only 0 or 1 values
+        ck.update((torch.randint(0, 2, size=(10, 2, 3)).long(), torch.arange(0, 10).long()))
+
+    with pytest.raises(ValueError):
+        # y_pred values are not thresholded to 0, 1 values
+        ck.update(
+            (
+                torch.rand(
+                    10,
+                ),
+                torch.randint(0, 2, size=(10, 3, 1)).long(),
+            )
+        )
+
+    with pytest.raises(ValueError):
+        # incompatible shapes
+        ck.update((torch.randint(0, 2, size=(10, 2, 3)).long(), torch.randint(0, 2, size=(10, 5)).long()))
+
+    with pytest.raises(ValueError):
+        # incompatible shapes
+        ck.update((torch.randint(0, 2, size=(10, 5, 6)).long(), torch.randint(0, 2, size=(10,)).long()))
+
+    with pytest.raises(ValueError):
+        # incompatible shapes
+        ck.update((torch.randint(0, 2, size=(10,)).long(), torch.randint(0, 2, size=(10, 5, 6)).long()))
+
+
+def test_check_shape():
+    ck = CohenKappa()
+
+    with pytest.raises(ValueError):
+        ck._check_shape((torch.randint(0, 2, size=(10, 1, 5, 12)).long(), torch.randint(0, 2, size=(10, 5, 6)).long()))
+
+    with pytest.raises(ValueError):
+        ck._check_shape((torch.randint(0, 2, size=(10, 1, 6)).long(), torch.randint(0, 2, size=(10, 5, 6)).long()))
+
+    with pytest.raises(ValueError):
+        ck._check_shape((torch.randint(0, 2, size=(10, 1)).long(), torch.randint(0, 2, size=(10, 5, 2)).long()))
+
+
+def test_cohen_kappa_non_weighted():
```

Comment:
Let's try to use `pytest.mark.parametrize` here. For example, take a look here : https://github.com/pytorch/ignite/pull/1683/files#diff-be88f17b02401b784cf4081d7e84bc7470d9beac1ac42294b8a0b1fc731900b3R20

Such that we can refactor all those 3 almost idenitcal functions. Probably, a similar thing can be done for test_integration_cohen_kappa_non_weighted_with_output_transform and its 2 triplets...

Reply:
I understood how it works. I can use `test.mark.parametrize` to replace all of these function with just 2 or 3. Should i do that?

Reply:
> I understood how it works. I can use test.mark.parametrize to replace all of these function with just 2 or 3. Should i do that?

The idea is to make test code better with minimal code duplication:
3 test functions -> 1 test function with parametrize

- `test_cohen_kappa_*` -> `test_cohen_kappa`
-  `test_integration_cohen_kappa_*_with_output_transform` -> `test_integration_cohen_kappa_with_output_transform`

Sorry, I do not quite get your question...

Reply:
@vfdev-5, Updated all the tests, and added new tests.

Reply:
@vfdev-5, It breaks the CI. The issue is the code formatting, `black` and `pycodestyle` don't agree on whitespaces before the columns. Any idea how to fix this?

Reply:
Can you please put a link to where there is a failure between code formatters ?

Reply:
Aftert running `black test_cohen_kappa.py` the columns in lines 98, 107, 108 get reformatted and whitespaces added before the columns. But then `pycodestyle test_cohen_kappy.py` shows that i should remove the whitespaces before the columns. And that's how the unit tests in the CI fail

Reply:
@vfdev-5, should i change the logic in `update_fn` in `test_cohen_kappa_all_weights_with_output_transform` for avoid these CI errors?

Reply:
@KickItLikeShika I do not quite understand your problem with formatting here. I updated your branch to the master to see where it is failing.

Reply:
Looks like it is OK.

Reply:
Well! i will commit the latest updates now with and delete the tests you told my about